src/commands/stack/add.ts

```diff
@@ -0,0 +1,185 @@
+import { Command, Flags, ux } from "@oclif/core";
+import { input, confirm } from "@inquirer/prompts";
+import { Endpoint, ShellConfig } from "../../lib/config";
+import FaunaClient from "../../lib/fauna-client";
+import { searchSelect } from "../../lib/search-select";
+
+export default class AddStackCommand extends Command {
+  static flags = {
+    name: Flags.string({
+      description: "New stack name",
+    }),
+    endpoint: Flags.string({
+      description: "Endpoint to use in this stack",
+    }),
+    database: Flags.string({
+      description: "Database path to use in this stack",
+    }),
+    "non-interactive": Flags.boolean({
+      description: "Disables interaction",
+      dependsOn: ["name", "endpoint", "database"],
+    }),
+    "set-default": Flags.boolean({
+      description: "Sets this stack as the default",
+    }),
+  };
+
+  static description = "Adds a new stack to the `.fauna-project`.";
+
+  static examples = [
+    "$ fauna stack add",
+    "$ fauna stack add --name my-app --endpoint dev --database my-database",
+    "$ fauna stack add --name my-app --endpoint dev --database my-database --set-default",
+  ];
+
+  async run() {
+    const config = ShellConfig.read({});
+
+    await this.execute(config);
+  }
+
+  async execute(config: ShellConfig) {
+    const { flags } = await this.parse();
+
+    if (config.projectConfig === undefined) {
+      this.error(
+        "No `.fauna-project` found. Create one with `fauna project init`."
+      );
+    }
+
+    const endpointName =
+      flags.endpoint ??
+      (await searchSelect({
+        message: "Select an endpoint",
+        choices: Object.keys(config.rootConfig.endpoints).map((endpoint) => ({
+          value: endpoint,
+        })),
+      }));
+    if (!Object.keys(config.rootConfig.endpoints).includes(endpointName)) {
+      this.error(`No such endpoint '${endpointName}'`);
+    }
+
+    let databaseName: string =
+      flags.database ??
+      (await this.promptDatabasePath(
+        config.rootConfig.endpoints[endpointName]
+      ));
+
+    const validateStackName = (input: string) => {
+      if (input.length === 0) {
+        return "Stack name cannot be empty";
+      } else if (
+        Object.keys(config.projectConfig?.stacks ?? {}).includes(input)
+      ) {
+        return `Stack ${input} already exists`;
+      } else {
+        return true;
+      }
+    };
+
+    const name =
+      flags.name ??
+      (await input({
+        message: "Stack name",
+        validate: validateStackName,
+      }));
+    const res = validateStackName(name);
+    if (res !== true) {
+      this.error(res);
+    }
```

Comment:
since we are passing that method as a validate function to their input method, do we need to validate again?

Reply:
we validate again in case they passed a command line argument